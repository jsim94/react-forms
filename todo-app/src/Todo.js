function Todo({ content, funcs, id }) {
  const handleEdit = (evt) => {
    const idx = evt.currentTarget.dataset.target;

    funcs.startEdit(idx);
  };

  const handleDelete = (evt) => {
    const id = evt.currentTarget.dataset.target;
    funcs.delTodo(id);
  };

  return (
    <div className="Todo">
      <p>{content}</p>

      <button onClick={handleEdit} data-target={id}>
        Edit
      </button>
      <button onClick={funcs.handleDelete} data-target={id}>
        Remove
      </button>
    </div>
  );
}

export default Todo;
