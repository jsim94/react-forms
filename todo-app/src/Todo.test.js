import { render, fireEvent } from "@testing-library/react";
import Todo from "./Todo";

it("should render", function () {
  render(<Todo funcs={{ handleDelete: null }} />);
});

it("should match snapshot", function () {
  const { asFragment } = render(
    <Todo content="Test todo" funcs={{ handleDelete: null }} />
  );
  expect(asFragment()).toMatchSnapshot();
});
