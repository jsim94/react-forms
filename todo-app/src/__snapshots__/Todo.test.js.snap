// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <p>
      Test todo
    </p>
    <button>
      Edit
    </button>
    <button>
      Remove
    </button>
  </div>
</DocumentFragment>
`;
